core/market/decentralized/src/db/dao/offer.rs

```diff
@@ -48,6 +49,18 @@ impl<'c> OfferDao<'c> {
         .await
     }
 
+    pub async fn get_offers(&self, node_id: Option<NodeId>) -> DbResult<Vec<Offer>> {
+        readonly_transaction(self.pool, move |conn| {
+            Ok(match node_id {
+                Some(ident) => dsl::market_offer
+                    .filter(dsl::node_id.eq(ident))
+                    .load::<Offer>(conn)?,
+                _ => dsl::market_offer.load::<Offer>(conn)?,
```

Comment:
It is still not compatible with [spec](https://github.com/golemfactory/ya-client/blob/59684245250a66b35a495914048f55f2dd8a24ce/specs/market-api.yaml#L47-L50):
> Fetches all **active** Offers which have been published by the Provider

**Active** Offer means it is not expired, nor unsubscribed.

I've already done sth very similar in #333
https://github.com/golemfactory/yagna/pull/333/files#diff-e37852b488da770202f59605ea52db90R64-R82